[IV-25-26] Objetivo 3

gabrielherreraloz opened the pull request:
* [x] ¿Incluye el fichero `iv.yaml` las claves de todos los objetivos
anteriores?
* [x] ¿Se ha desarrollado el fichero para las tareas a partir de un
   issue?
* [x] ¿Ese issue está asignado a un milestone?
* [x] ¿El milestone incluye, entre sus productos mínimamente viables, la
  existencia de herramientas que permitan automatizar tareas?
* [x] ¿Hemos justificado correctamente en caso de elegir un gestor de
   dependencias el que se use como gestor de tareas?
* [x] ¿El PR, que también es un issue, está asignado al mismo milestone?
* [x] ¿La tarea `check` efectivamente comprueba la sintaxis de los fuentes
  existentes y sólo eso?
* [x] ¿Se ha comprobado que las instrucciones incluidas en el `README.md` son
completas y correctas?

Closes #13 
Closes #15

chelunike committed:
Completando ignore para IDE

chelunike committed:
Añadido descripción del lenguaje closes #7

chelunike committed:
Añadido modelo del problema closes #7

chelunike committed:
Fix typo in language key in iv.yaml

gabrielherreraloz committed:
#13 Añadida especificación de gestores

gabrielherreraloz committed:
Fix: Enlace erroneo en README.md

gabrielherreraloz committed:
Mejorada especificación de gestores(#13)

Se ha justificado mejor la elección de los gestores de tareas y deopendencias y se han indicado las distintas opciones propuestas

gabrielherreraloz committed:
Creado makefile con automatizaciones básicas (#15)

gabrielherreraloz committed:
Extendida documentación de los gestores (#13)

Se ha documentado más extensamente el motivo de la elección del gestor de tareas y se han especificado los criterios seguidos para ello.

gabrielherreraloz commented:
@gosema echale un vistazo porfa

gosema commented on docs/Gestores.md:
```diff
@@ -0,0 +1,70 @@
+# Decisiones tomadas en la elección de gestores enfocados en el uso de tests
+
+## Gestor de Dependencias
+Para este proyecto escrito en Go, se ha seleccionado como gestor de dependencias "Go Modules", el cual ha sido elegido ya que está directamente integrado en las herramientas base de Go y no es necesario instalar nada adicional, lo cual facilita la portabilidad del proyecto. Además, al ser el gestor nativo del lenguaje, lo hace perfectamente compatible con este. Además, Go Modules es sin duda el g ...
```
¿En base a que criterios usas go? En el título pones algo sobre que este enfocado a tests. ¿Es ese tu criterio?  y si lo fuese ¿lo mencionas en la justificación?

gabrielherreraloz replied:
Sí, en el título esta más orientado lo de los tests al gestor de tareas, en este caso que es el de dependencias, he escrito en general que es por facilidad de no instalar herramientas adicionales ya que con instalar Go viene incluido el gestor de dependencias y que lo hace muy compatible con el lenguaje porque es su gestor nativo.

gosema replied:
Entonces si justificas la elección guiándote por los tests ¿porque en la justificación no hablas de una explicación de que ventaja tiene Go modules sobre test?
Tambien conviene que indiques antes unos criterios por el que has elegido Go modules. Me explico mejor, dice scorrectamente porque lo utilizas pero no declaras antes que criterios te han hecho elegirlo.

gabrielherreraloz replied:
Vle, lo hago como en las tareas y dejo bien definidos los criterios.

gosema commented on docs/Gestores.md:
```diff
@@ -0,0 +1,70 @@
+  
+  - **Integración con el lenguaje**: Una alta integración permite tareas más complejas que hacerlas en shell ccomo interactuar con librerías o estructuras de datos del proyecto.
+
+  - **Confiabilidad**: El gestor seleccionado debe contar con mantenimiento activo, y que asegure su compatibilidad en versiones futuras.
```
¿Con qué mides que un gestor de tareas es más seguro que otro? ¿Es objetivo?

gabrielherreraloz replied:
Es cierto que no puede medirse de forma 100% objetiva como un numero exacto, así que elimino este criterio directamente.

gabrielherreraloz committed:
FIX - Eliminado un criterio en respuesta al comentario de @gosema

Se ha eliminado el criterio de la confiabilidad debido a que no es lo suficientemente objetivo.

gosema commented on docs/Gestores.md:
```diff
@@ -0,0 +1,62 @@
+
+  - **Compilación propia**: Si el gestor de tareas incluye un compilador propio, esto ayuda a la detección de errores antes de la ejecución del código, permite la reutilización directa del código del lenguaje nativo del proyecto, en este caso Go, además se evita la necesidad de utilizar herramientas externas.
+  
+  - **Integración con el lenguaje**: Una alta integración permite tareas más complejas que hacerlas en shell ccomo interactuar con librerías o estructuras de datos del proyecto.
```
Te basas en la complejidad para realiza tareas. ¿Realmente es algo objetivo? 
Por ponerte un ejemplo, un programador experto en shell que nunca ha tocado Go va a verlo de tu msima manera?

gabrielherreraloz replied:
Con complejas me refiero a que no son los comandos base tipo build o run, cambiaré el termino porque lo he espresado de manera que parece algo subjetivo.

gosema commented on docs/Gestores.md:
```diff
@@ -0,0 +1,62 @@
+
+ - **makefile**: Uno de los más conocidos y muy trabajado en la carrera de Ingeniería Informática. Cumplimiento de criterios:
+
+   - **Requisito de instalación**: Está instalado en Linux y MacOS de manera predeterminada por lo que no es necesaria su instalación independiente para su uso.
```
¿Y para un usuario de Windows?

gabrielherreraloz replied:
No, aunque es cierto que debería especificarlo mejor.

gosema reviewed (commented):
Buenas noches @gabrielherreraloz . Te he comentado un par de cosas que desde mi punto de vista se les puede dar una vuelta para lograr este objetivo

gabrielherreraloz committed:
Mejorada la documentación de los gestores

Se ha mejorado y matizado la documentación referida a los gestores de tareas y dependencias siguiendo los comentarios realizaados por @gosema

gosema commented on docs/Gestores.md:
```diff
@@ -0,0 +1,67 @@
+# Decisiones tomadas en la elección de gestores enfocados en el uso de tests
```
Sigo sin ver porque está enfocado en el uso de test. ¿Lo estas usando como criterio?¿Explicas como funciona esa herramienta con tests?

gosema commented on docs/Gestores.md:
```diff
@@ -0,0 +1,67 @@
+  - **Requisito de instalación**: Es preferible que el gestor de tareas elegido esté preinstalado de forma automática en el sistema o inlcuido en la propia instalación para utilizar el lenguaje del proyecto.
+
+### Decisión final
+Para este proyecto escrito en Go, se ha seleccionado como gestor de dependencias "Go Modules", el cual ha sido elegido ya que está directamente integrado en las herramientas base de Go y no es necesario instalar nada adicional, lo cual facilita la portabilidad del proyecto. Además, al ser el gestor nativo del lenguaje, lo hace perfectamente compatible con este. Además, Go Modules es sin duda el g ...
```
Estas explicando que es popular. ¿Es esto objetivo?

gosema commented on docs/Gestores.md:
```diff
@@ -0,0 +1,67 @@
+### Criterios de selección
+  - **Requisito de instalación**: Es preferible que el gestor de tareas elegido esté preinstalado de forma automática en el sistema o inlcuido en la propia instalación para utilizar el lenguaje del proyecto.
+
+### Decisión final
```
Puede ser que tengas claro desde un primer momento que gestor de dependencias utilizar. ¿Pero no deberías al menos utilizar otra herramienta posible con la que compararla?

gabrielherreraloz committed:
Mejorada la justificación del gestor de dependencias

Se ha mejorado y matizado la documentación referida al gestor dependencias siguiendo los comentarios realizados por @gosema

gosema commented on docs/Gestores.md:
```diff
@@ -0,0 +1,63 @@
+## Gestor de tareas
+
+ ### Criterios de selección
+  - **Requisito de instalación**: Es preferible que el gestor de tareas elegido esté preinstalado de forma automática en el sistema para así facilitar y aumentar su escalabilidad.
```
No entiendo que tiene que ver los requisitos de instalación con la escalabilidad

gabrielherreraloz replied:
Me he expresado mal, quería referirme a portabilidad, no escalabilidad

gosema commented on docs/Gestores.md:
```diff
@@ -0,0 +1,63 @@
+
+  - **Compilación propia**: Si el gestor de tareas incluye un compilador propio, esto ayuda a la detección de errores antes de la ejecución del código, permite la reutilización directa del código del lenguaje nativo del proyecto, en este caso Go, además se evita la necesidad de utilizar herramientas externas.
+  
+  - **Integración con el lenguaje**: Una alta integración permite tareas más allá que lo que permite shell, como interactuar con librerías o estructuras de datos del proyecto.
```
Sigo sin verle mucho sentido a este criterio, ¿acaso shell no puede realizar todas las tareas? Como desarrollador ¿qué ventajas te está dando este criterio?

gabrielherreraloz replied:
Hacerlas en Go directamente siendo el lenguaje nativo permite hacer tareas más avanzadas y mas eficientes, sin embargo lo voy a eliminar como criterio porque como digo en la elección final, solo voy a usar en principio tareas básicas.

gosema commented on docs/Gestores.md:
```diff
@@ -0,0 +1,63 @@
+
+  ### Elección final
+  
+  Finalemente, en base a los criterios mencionados, se ha decididio usar Makefile puesto que aporta mayor universalidad al estar instalado de forma predeterminada en muchos sistemas operativos, además su largo bagaje con años de mantenimiento elimina una posible deuda técnica generada a largo plazo. Sin bien es cierto que Goyek aporta mayor integración con Go en programación más avanzada, para la ...
```
De todos los criterios que tienes se podría decir que Goyek gana en 3 y make en solo 1. ¿Porqué te decantas por make cuando goyek te presenta más ventajas según tus criterios?
¿Es realmente objetivo que uses make finalmente por sus años de mantenimiento ¿¿Es realmente esto un criterio establecido?

gabrielherreraloz committed:
Modificados los criterios del gestor de tareas

Se han eliminado los criterios 2 y 3 (compilador propio e integración con el lenguaje) debido a que analizandolos bien, podrían ser beneficiosos en un proyectop de mayor magnitud, pero en nuestro caso no son tan relevantes

gosema commented on docs/Gestores.md:
```diff
@@ -0,0 +1,47 @@
+
+  ### Elección final
+  
+  Finalemente, en base a los criterios mencionados, se ha decididio usar Makefile puesto que aporta mayor portabilidad al estar instalado de forma predeterminada en muchos sistemas operativos, además su largo bagaje con años de mantenimiento elimina una posible deuda técnica generada a largo plazo. Si bien es cierto que Goyek se escribe en Go al igual que el proyecto, no justifica su elección por ...
```
Sobre su mantenimiento, no esta puesto como criterio y no creo que te aporte nada a tomar la decisión. Es ruido

gosema commented on docs/Gestores.md:
```diff
@@ -0,0 +1,47 @@
+
+## Gestor de Dependencias
+
+Para este proyecto escrito en Go, se ha seleccionado como gestor de dependencias "Go Modules", es realemnete la unica posibilidad que se plantea ya que "Go Modules" es la única versión realista y actualizada a día de hoy, puesto que el resto de posibilidades están obsoletas y todos proyectos de Go en la actualidad usan esta herramienta, siendo además "Go Modules" el gestor de dependencias oficial ...
```
Quizá convendría añadir un criterio de elección aquí. Aunque no haya opciones actuales viables y así estableces otro argumento de peso por el que has elegido el lenguaje go. Me explico, de esta forma solo parece que lo utilizas porque es popular y no te queda más remedio.

gosema reviewed (commented):
Buenas noches @gabrielherreraloz , veo que tu trabajo ha mejorado bastante. Considera estas observaciones que te he hecho y tienes mi visto bueno

gabrielherreraloz committed:
Se han matizado las jutificaciones de las elecciones tomadas (#13)

Se ha mejorado la redacción y especificado mejor los criterios de selección siguiendo los comentarios realizados por @gosema

gosema reviewed (approved):
Creo que tu documentación sobre los gestores ha mejorado bastante, ahora con criterios objetivos que te permiten elegir una herramienta adecuada para tu proyecto. Por mi parte tiene el visto bueno

gabrielherreraloz commented:
@JJ listo para revisión

JJ commented on docs/Gestores.md:
```diff
@@ -0,0 +1,51 @@
+ ### Criterios de selección
+  - **Requisito de instalación**: Es preferible que el gestor de tareas elegido esté preinstalado de forma automática en el sistema para así facilitar y mejorar su portabilidad entre sistemas.
+
+  - **Lenguaje de ejecución**: Si el lenguaje usado por el gestor coincide con el lenguaje elegido para el proyecto, facilita en gran medida su utilización.
```
¿Por qué lo facilita? Recuerda que cosas como "facilitar" no se pueden medir objetivamente.

JJ commented on docs/Gestores.md:
```diff
@@ -0,0 +1,51 @@
+ ### Opciones posibles
+Para la elección del gestor de tareas se presentaban distintas opciones bastante interesantes, de las cuales he querido recalcar tres:
+
+ - **GoTask**: Se ha tenido en cuenta por haber sido referenciada por algunos usuarios de Reddit. Cumplimiento de criterios:
```
Por favor, enlazad lo que proponéis.

JJ commented on docs/Gestores.md:
```diff
@@ -0,0 +1,51 @@
+   - **Lenguaje de ejecución**: Shell, comandos de ordenes del sistema.
+ 
+
+ - **Goyek**: También muy referenciado y más recomendado que gotaskr en Reddit. Cumplimiento de criterios:
```
Estás mencionando un criterio que no has incluido. Si está muy o poco referenciado tienes que tratar que sea objetivo y también dónde lo has visto, "visto en reddit" es muy genérico.

JJ commented on docs/Gestores.md:
```diff
@@ -0,0 +1,51 @@
+   
+   - **Requisito de instalación**: No viene instalado de forma predeterminada en ningún sistema operativo por lo que es necesario la instalación del binario task.
+   
+   - **Lenguaje de ejecución**: Shell, comandos de ordenes del sistema.
```
No sé muy bien por qué pones esto. Tampoco es que sea malo que para ejecutar tareas del sistema se usen comandos del sistema, ¿no?

JJ reviewed (commented):
Aparte de lo dicho, cambia la base al objetivo 2 para que no se vean todos los ficheros que hay en ese objetivo.

gabrielherreraloz commented:
> Aparte de lo dicho, cambia la base al objetivo 2 para que no se vean todos los ficheros que hay en ese objetivo.

Ya lo intenté, no me sale la opción de hacerlo sobre el objetivo 2, solo me salen sobre el 1 y el 0.

gabrielherreraloz committed:
Matizada la explicación de los criterios (#13)

Siguiendo los comentarios realizados por @JJ se han explicado mas claramente los criterios del lenguaje de ejecución y se han eliminado las breves descripciones iniciales de cada opción porque son inexactas, se han añadido en su lugar las documentaciones correspondientes.

gabrielherreraloz commented:
@JJ he intentado explicar mejor lo que has comentado, si siguiera sin estar demasiado objetivo o claro, lo quito directamente.

JJ commented:
Invalida la revisión de gosema; cuando él lo mire ya le doy yo una vuelta.

gosema commented on docs/Gestores.md:
```diff
@@ -0,0 +1,51 @@
+## Gestor de tareas
+
+ ### Criterios de selección
+  - **Requisito de instalación**: Es preferible que el gestor de tareas elegido esté preinstalado de forma automática en el sistema para así facilitar y mejorar su portabilidad entre sistemas.
```
Un error parecido al que te comentó JJ, ¿Por qué lo facilita? Recuerda que cosas como "facilitar" no se pueden medir objetivamente.

gosema reviewed (commented):
No has cambiado la base del proyecto al objetivo 2. Esto nos facilitaría la correccion y los archivos que muestra como cambiados

gabrielherreraloz commented:
> No has cambiado la base del proyecto al objetivo 2. Esto nos facilitaría la correccion y los archivos que muestra como cambiados

Lo sé, pero no puedo hacerlo.

<img width="332" height="293" alt="Captura desde 2025-11-25 12-29-47" src="https://github.com/user-attachments/assets/d3cdb415-a206-4055-9dcf-eb8b61b6ec6d" />

gabrielherreraloz committed:
Simplificada explicación de criterio (#13)

Siguiendo el comentario de @gosema, he eliminado directamente la palabra "facilita" del criterio de instalación por ser poco objetiva.

gosema commented:
> > No has cambiado la base del proyecto al objetivo 2. Esto nos facilitaría la correccion y los archivos que muestra como cambiados
> 
> Lo sé, pero no puedo hacerlo.
> 
> <img alt="Captura desde 2025-11-25 12-29-47" width="332" height="293" src="https://private-user-images.githubusercontent.com/162113210/518613416-d3cdb415-a206-4055-9dcf-eb8b61b6ec6d.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NjQwNzA1NjgsIm5iZiI6MTc2NDA3MDI2OCwicGF0aCI6Ii8xNjIxMTMyMTAvNTE4NjEzNDE2LWQzY2RiNDE1LWEyMDYtNDA1NS05ZGNmLWViOGI2MWI2ZWM2ZC5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjUxMTI1JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI1MTEyNVQxMTMxMDhaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT01YTJlMzZkZGI2N2Q1MGYyYjY5N2JjMzcyNWQ1NmNjNjNiOGU1NWFiZGRjZmM3ODc3YzBiNjdjY2U4NmQ2ZTI0JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.S2LpXxPZZkHK16NgfskTK1wMimRxtdJKPlEFP7m1p9U">

Creo que esto se esta produciendo porque chelunike esta subiendo el objetivo desde su rama de su repositorio a tu trabajo.
Creo que github no te permite cambiar la base a un fork si ya has creado el pull request, lo único que se me ocurre es que cierres el PR y al crear uno nuevo PR darle a la opción de "compare across forks" donde podrás elegir el fork de tu compañero.
De todas formas creo que en este caso convendría dejarl tu proyecto como esta y tener en cuenta esto para futuros objetivos.

Sobre los cambios que has hecho de las observaciones de JJ están correctos. Te vuelvo a aprobar mi revisión

gabrielherreraloz commented:
@JJ Listo de nuevo

JJ commented:
Todavía no lo ha aparobado @gosema , ¿no?

gosema commented:
> Todavía no lo ha aparobado @gosema , ¿no?

Lo acabo de aprobar perdón, se me olvidó darle a submit review

JJ commented on docs/Gestores.md:
```diff
@@ -0,0 +1,51 @@
+ ### Criterios de selección
+  - **Requisito de instalación**: Es preferible que el gestor de tareas elegido esté preinstalado de forma automática en el sistema para así mejorar su portabilidad entre sistemas.
+
+  - **Lenguaje de ejecución**: Si el lenguaje usado por el gestor coincide con el lenguaje elegido para el proyecto, permite a este trabajar sobre las funciones, clases o librerias del código directamente. Además, al usar el mismo lenguaje, no son necesarios intérpretes externos para este, simplificando la instalación y ejecución.
```
Aunque objetivo, esto es absolutamente arbitrario, aparte de que "el lenguaje usado" no está claro. ¿Se trata del lenguaje en el que está escrito? ¿Qué más da si está compilado? ¿Se trata del lenguaje para las tareas? ¿Por qué quieres que esté en el mismo lenguaje si la mayorái de las tareas van a ser de la línea de órdenes?

gabrielherreraloz replied:
Voy a quitar este criterio y dejar solamente el primero y ya está.

JJ commented on docs/Gestores.md:
```diff
@@ -0,0 +1,51 @@
+
+   - **Requisito de instalación**: Está instalado en Linux y MacOS de manera predeterminada por lo que no es necesaria su instalación independiente para su uso si es en uno de estos sistemas, cosa que no se cumple si lo hacemos en Windows.
+
+   - **Lenguaje de ejecución**: Shell, comandos de ordenes del sistema. En la mayoría de sistemas no supone un problema pero en algunos entornos como Windows podría ser necesario instalar un intérprete adicional.
```
En Windows usa el lenguaje que quieras. De hecho, creo que en Make se puede cambiar https://unix.stackexchange.com/questions/217243/which-shell-is-used-in-gnu-make-files

JJ commented on docs/Gestores.md:
```diff
@@ -0,0 +1,51 @@
+
+  ### Elección final
+  
+  Finalemente, en base a los criterios mencionados, se ha decididio usar Makefile puesto que aporta mayor portabilidad al estar instalado de forma predeterminada en muchos sistemas operativos. Si bien es cierto que Goyek se escribe en Go al igual que el proyecto, no justifica su elección por delante del requisito de instalación, que prevalece en importancia al marcar una diferencia mucho más sign ...
```
la orden se llama `make`, el comando Makefile. No veo que aquí tenga absolutamente nada que ver el segundo criterio, que sigue siendo arbitrario.

JJ commented on docs/Gestores.md:
```diff
@@ -0,0 +1,51 @@
+
+### Comandos de interés de makefile
+  - "make build": compila el pryecto.
+  - "make install deps": instala las dependencias necesarias con Go Modules.
```
¿Algo en el guión pide esto?
¿Aparte, `install deps` es un comando o dos?

JJ reviewed (commented):
Tratad de evitar contenidos que no sean relevantes.

gabrielherreraloz committed:
Se ha eliminado el criterio del lenguaje (#13)

Se ha eliminado el criterio del lenguaje del gestor de dependencias al no ser del todo relevante y conseguir presentarlo como un criterio 100% objetivo y definido.

gabrielherreraloz committed:
FIX-Eliminados archivos del Objetivo 2 para mayor claridad en este, se hará merge cuando se haga el del objetivo 2 para no generar conflictos.

gosema reviewed (approved):
Has entendido y aplicado correctamente los cambios que te ha indicado JJ. Lo veo bien paar revisión.

gabrielherreraloz commented:
@JJ Listo

JJ commented on Makefile:
```diff
@@ -0,0 +1,23 @@
+
+check:
+	go fmt ./...
+	go vet ./...
```
¿Cuál de las dos comprueba la sintaxis?

JJ commented on Makefile:
```diff
@@ -0,0 +1,23 @@
+	go fmt ./...
+	go vet ./...
+
+clean:
```
Siempre tienes que basarte en el producto mínimo. No crees otros objetivos que no se incluyan, porque hay de todas formas que revisarlos.

JJ commented on Makefile:
```diff
@@ -0,0 +1,23 @@
+	@go build -o "${BIN_FILE}"
+	
+run:
+	./"${BIN_FILE}"
```
Cuando se ejecute, no se va a ejecutar un fichero binario. Una vez más, solo objetivos incluidos en el hito.

JJ reviewed (commented):
Tened *siempre* en cuenta el concepto de producto *mínimo*.

gabrielherreraloz committed:
Se han eliminado bloques de tareas no aplicables al PMV(#15)

JJ reviewed (approved):
👍

gabrielherreraloz committed:
Eliminados más bloques no aplicables al PMV(#15)
